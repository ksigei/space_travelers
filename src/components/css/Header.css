.navbox {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 17vh;
  align-items: center;
  border-bottom: 1px solid #e4e4e4;
  margin-bottom: 20px;
}

.icon{
  display: flex;
  align-items: center;
  text-decoration: none;
}

.icon h2{
  margin-left: 20px;
  color: #000;
}

nav .icon img {
  height: 15vh;
}

.link{
  text-decoration: none;
  margin: 0 15px;
}

.link.active{
  text-decoration: underline;
}
