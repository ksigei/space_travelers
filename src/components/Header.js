import React from 'react';
import { Link } from 'react-router-dom';
import logoUrl from '../assets/planet.png';
import './css/Header.css';

function Header() {
  return (
    <nav>
      <Link className="icon" to="/">
        <img src={logoUrl} alt="planet" />
      </Link>
      <Link to="/missions">Missions</Link>

    </nav>
  );
}

export default Header;
