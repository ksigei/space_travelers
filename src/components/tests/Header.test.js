import { render, screen, fireEvent } from '@testing-library/react';
import { HashRouter } from 'react-router-dom';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import Header from '../Header';
import '@testing-library/jest-dom';

expect.extend({ toMatchDiffSnapshot });

test('@Header Rocket Navigation', () => {
  const { asFragment } = render(<HashRouter><Header /></HashRouter>);
  fireEvent.click(screen.getByText(/Rockets/));
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Rockets/));

  expect(firstRender).toMatchDiffSnapshot(asFragment());
});

test('@Header Mission Navigation', () => {
  const { asFragment } = render(<HashRouter><Header /></HashRouter>);
  fireEvent.click(screen.getByText(/Missions/));
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Missions/));
  expect(firstRender).toMatchDiffSnapshot(asFragment());
});

test('@Header Profile Navigation', () => {
  const { asFragment } = render(<HashRouter><Header /></HashRouter>);
  fireEvent.click(screen.getByText(/Profile/));
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Profile/));
  expect(firstRender).toMatchDiffSnapshot(asFragment());
});
