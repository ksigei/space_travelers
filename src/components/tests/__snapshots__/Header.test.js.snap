// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@Header Mission Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`@Header Profile Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`@Header Rocket Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
